src/java/com/wolfssl/provider/jsse/WolfSSLSocket.java

```diff
@@ -1783,8 +1783,13 @@ public int read(byte[] b, int off, int len)
                 }
 
                 /* do handshake if not completed yet, handles synchronization */
-                if (socket.handshakeComplete == false) {
-                    socket.startHandshake();
+                try {
+                    /* do handshake if not completed yet, handles synchronization */
+                    if (socket.handshakeComplete == false) {
+                        socket.startHandshake();
+                    }
+                } catch (SocketTimeoutException e) {
+                    throw new SocketTimeoutException("ssl.read in startHandshake");
```

Comment:
Same as above, should we add a debug log for the message, then `throw e` for the original exception?